ratis-netty/src/main/java/org/apache/ratis/netty/client/NettyClientStreamRpc.java

```diff
@@ -118,14 +121,23 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
         }
         final DataStreamReply reply = (DataStreamReply) msg;
         LOG.debug("{}: read {}", this, reply);
-        ClientInvocationId clientInvocationId = ClientInvocationId.valueOf(reply.getClientId(), reply.getStreamId());
+        clientInvocationId = ClientInvocationId.valueOf(reply.getClientId(), reply.getStreamId());
         Optional.ofNullable(replies.get(clientInvocationId))
             .map(Queue::poll)
             .ifPresent(f -> f.complete(reply));
       }
 
       @Override
       public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+        Optional.ofNullable(clientInvocationId).ifPresent(id ->
+            Optional.ofNullable(replies.get(id)).ifPresent(q -> {
+              for (int i = 0; i < q.size(); i++) {
+                CompletableFuture<DataStreamReply> f = q.poll();
+                f.completeExceptionally(cause);
+              }
+              replies.remove(id);
+            })
+        );
```

Comment:
We should use replies.remove(id) instead of replies.get(id).  Also, the code can be simplified as below:
```
        Optional.ofNullable(clientInvocationId)
            .map(replies::remove)
            .orElseGet(LinkedList::new)
            .forEach(f -> f.completeExceptionally(cause));
```

Reply:
Has been modified @szetszwo